extensions/common/iam/verifiable-credentials/src/test/java/org/eclipse/edc/iam/verifiablecredentials/StatusList2021RevocationServiceTest.java

```diff
@@ -110,4 +109,58 @@ void checkRevocation_whenCached_valid() {
+        var credential = TestFunctions.createCredentialBuilder().credentialStatus(new CredentialStatus("test-id", "StatusList2021",
+                        Map.of(STATUS_LIST_PURPOSE, "revocation",
+                                STATUS_LIST_INDEX, REVOKED_INDEX,
+                                STATUS_LIST_CREDENTIAL, "http://localhost:%d/credentials/status/3".formatted(clientAndServer.getPort()))))
+                .build();
+        assertThat(revocationService.getStatusPurpose(credential)).isSucceeded()
+                .isEqualTo("revocation");
+    }
+
+    @Test
+    void getStatusPurposes_whenMultipleCredentialStatusRevoked() {
+        clientAndServer.reset();
+        clientAndServer.when(request().withMethod("GET").withPath("/credentials/status/3"))
+                .respond(HttpResponse.response().withStatusCode(200).withBody(TestData.STATUS_LIST_CREDENTIAL_SUBJECT_IS_ARRAY));
+        var credential = TestFunctions.createCredentialBuilder().credentialStatus(new CredentialStatus("test-id", "StatusList2021",
+                        Map.of(STATUS_LIST_PURPOSE, "revocation",
+                                STATUS_LIST_INDEX, REVOKED_INDEX,
+                                STATUS_LIST_CREDENTIAL, "http://localhost:%d/credentials/status/3".formatted(clientAndServer.getPort()))))
+                .build();
+        assertThat(revocationService.getStatusPurpose(credential)).isSucceeded()
+                .isEqualTo("revocation");
+    }
+
+    @Test
+    void getStatusPurpose_whenCredentialStatusNotActive() {
+        clientAndServer.reset();
+        clientAndServer.when(request().withMethod("GET").withPath("/credentials/status/3"))
+                .respond(HttpResponse.response().withStatusCode(200).withBody(TestData.STATUS_LIST_CREDENTIAL_SINGLE_SUBJECT));
+        var credential = TestFunctions.createCredentialBuilder().credentialStatus(new CredentialStatus("test-id", "StatusList2021",
+                        Map.of(STATUS_LIST_PURPOSE, "revocation",
+                                STATUS_LIST_INDEX, NOT_REVOKED_INDEX,
+                                STATUS_LIST_CREDENTIAL, "http://localhost:%d/credentials/status/3".formatted(clientAndServer.getPort()))))
+                .build();
+        assertThat(revocationService.getStatusPurpose(credential)).isSucceeded()
+                .isNull();
+    }
+
+    @Test
+    void getStatusPurpose_whenNoCredentialStatus() {
+        clientAndServer.reset();
+        clientAndServer.when(request().withMethod("GET").withPath("/credentials/status/3"))
+                .respond(HttpResponse.response().withStatusCode(200).withBody(TestData.STATUS_LIST_CREDENTIAL_SINGLE_SUBJECT));
+        var credential = TestFunctions.createCredentialBuilder().build();
+        assertThat(revocationService.getStatusPurpose(credential))
+                .isSucceeded()
+                .isNull(); //content is null, not the result!
```

Comment:
adding a `.isNotNull()` before `.isSucceeded()` could remove the need for the comment